### PR TITLE
API GET - Filter articles marked as "featured"

### PR DESCRIPTION
**Pull request by carlitorweb**
## Summary of Changes
Adding 'featured' articles filter to the articles controller webservice-api

## Testing Instructions
Make a GET call to .....api/v1/content/articles?filter[featured]=1

## Actual result BEFORE applying this Pull Request 
You cant filter the response by featured articles

## Expected result AFTER applying this Pull Request 
Response with only the articles marked as features (featured=1)

## Link to documentations
Update https://docs.joomla.org/J4.x:Joomla_Core_APIs by adding a new endpoint: **/api/index.php/v1/content/articles?filter[featured]=1** Maybe add the others filters too?

### CODE REVIEW

File: api/components/com_content/src/Controller/ArticlesController.php [hunk 1]
```diff
@@ -68,6 +68,10 @@ public function displayList()
         if (\array_key_exists('state', $apiFilterInfo)) {
             $this->modelState->set('filter.published', $filter->clean($apiFilterInfo['state'], 'INT'));
         }
+        
+        if (\array_key_exists('featured', $apiFilterInfo)) {
```
Comment (richard67) [inline api/components/com_content/src/Controller/ArticlesController.php]:
```suggestion

        if (\array_key_exists('featured', $apiFilterInfo)) {
```
Remove white space in empty line to fix code style check failing in drone.

**Comment by carlitorweb** [comment]
@wilsonge @richard67  thank you for the edit

Wonder if is possible, now that  we can filtre by featured articles, can we somehow apply the ` fp.ordering`? will complete this GET

**Comment by ReLater** [comment]
Addition to the test instructions: For stupid people like me who use a Linux console and token authentication: you have to mask the square brackets for `filter` with backslashes. For example:

`curl -H "Authorization: Bearer MY_AUTH_TOKEN" "https://example.org/api/index.php/v1/content/articles?filter\[featured\]=1"`

I have absolutely no idea where to insert the `GET` if that's important for successful testing(?).

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on 45db7d68c5263fd038024d152cff253c01b428fc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39273">issues.joomla.org/tracker/joomla-cms/39273</a>.</sub>

**Comment by carlitorweb** [comment]
> Addition to the test instructions: For stupid people like me who use a Linux console and token authentication: you have to mask the square brackets for `filter` with backslashes. For example:
> 
> `curl -H "Authorization: Bearer MY_AUTH_TOKEN" "https://example.org/api/index.php/v1/content/articles?filter\[featured\]=1"`
> 
> I have absolutely no idea where to insert the `GET` if that's important for successful testing(?).



> Addition to the test instructions: For stupid people like me who use a Linux console and token authentication: you have to mask the square brackets for `filter` with backslashes. For example:
> 
> `curl -H "Authorization: Bearer MY_AUTH_TOKEN" "https://example.org/api/index.php/v1/content/articles?filter\[featured\]=1"`
> 
> I have absolutely no idea where to insert the `GET` if that's important for successful testing(?).

Thank for add the test instructions. 

About the `GET` was more like to be explicit, for example for those who use some vscode extension like Thunder Client for HTTP request where you need choose between GET, POST etc , Is not needed in your case

**Comment by ReLater** [comment]
@wilsonge 
Thanks for clarification

**Comment by alikon** [comment]
please see #38798 too

**Review by wilsonge — APPROVED** [review]

**Comment by alikon** [comment]
as #38798 has been merged this should be closed

**Pull request closed without merging**